src/Kdyby/Translation/Loader/DoctrineLoader.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace Kdyby\Translation\Loader;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Exception\TableNotFoundException;
+use Doctrine\DBAL\Schema\AbstractSchemaManager;
+use Doctrine\DBAL\Types\Type;
+use Doctrine\ORM\Mapping\Builder\ClassMetadataBuilder;
+use Doctrine\ORM\Tools\SchemaTool;
+use Kdyby\Doctrine\EntityManager;
+use Kdyby\Doctrine\Mapping\ClassMetadata;
+use Kdyby\Translation\Listener\TranslationMetadataListener;
+use Kdyby\Translation\Resource\DatabaseResource;
+use Tracy\Debugger;
+
+class DoctrineLoader extends DatabaseLoader
+{
+
+    /** @var Connection */
+    private $conn;
+
+    /**
+     * @param Connection $conn
+     */
+    public function __construct(Connection $conn)
+    {
+        $this->conn = $conn;
+    }
+
+    /**
+     * @return array
+     */
+    public function getLocales()
+    {
+        if( !function_exists( 'array_column' ) ) {                  //just because of PHP 5.4, where function array_column is not present. Fuck you, PHP 5.4
+            function array_column( array $input, $column_key, $index_key = null ) {
+                $result = array();
+                foreach( $input as $k => $v )
+                    $result[ $index_key ? $v[ $index_key ] : $k ] = $v[ $column_key ];
+                return $result;
+            }
+        }
+
+        $qb = $this->conn->createQueryBuilder()
+            ->addSelect("DISTINCT `$this->locale` AS locale")
+            ->from("`$this->table`");
+        try {
+            $stmt = $qb->execute();
+            $locales = array_column($stmt->fetchAll(), 'locale');
+        } catch(TableNotFoundException $e) {
+            $locales = array();
+        }
+        return $locales;
+    }
+
+
+    protected function getResourceName()
+    {
+        return DatabaseResource::DOCTRINE;
+    }
+
+    protected function getTranslations($locale)
+    {
+        $qb = $this->conn->createQueryBuilder()
+            ->addSelect("`$this->key` AS `key`")
```

Comment:
Jak se to bude chovat na jinych db nez mysql? mel bys asi pouzit nejaky `quoteIdentifier()`, to same u NDBT, tam je to nejak `getSupplementalDriver()->delimite()`